python/src/opendp/_convert.py

```diff
@@ -689,15 +686,17 @@ def optional_u32(ptr):
         max_influenced_partitions=optional_u32(ptr_data[5]),
     )
 
+def _check_polars_by(by):
+    if isinstance(by, str):
+        raise ValueError(f"by ({by}) must be a collection; Did you mean [\"{by}\"]?")
+    
+    if not isinstance(by, Sequence):
+        raise ValueError(f"by ({by}) must be a collection")
```

Comment:
Why "collection"? There are python collections (https://docs.python.org/3/library/collections.html), but that's not what you mean. Either "sequence type" or "list" would probably be better I think.